src/main/java/capstone/mfslbackend/service/FantasyLeagueService.java

```diff
@@ -145,35 +148,47 @@ public List<FantasyWeek> createFantasyLeagueSchedule(Long leagueId) {
             schedule.add(new ArrayList<>(matches));
 
         }
-        int w = 0;
-        // Create the weeks
-        for (int matchIndex = 0; matchIndex < matches.size(); matchIndex += 2) {
-            FantasyWeek fantasyWeek = new FantasyWeek();
-            FantasyTeam teamA = matches.get(matchIndex);
-            FantasyTeam teamB = matches.get(matchIndex + 1);
-
-
-            Set<FantasyWeek> weeksA = teamA.getFantasyWeeks();
-            Set<FantasyWeek> weeksB = teamB.getFantasyWeeks();
-            fantasyWeek.setFantasyTeamA(teamA);
-            fantasyWeek.setFantasyTeamB(teamB);
-            weeksA.add(fantasyWeek);
-            weeksB.add(fantasyWeek);
-            teamA.setFantasyWeeks(weeksA);
-            teamB.setFantasyWeeks(weeksB);
-
-            //fix order number once decided
-            teamA.setOrderNumber(matchIndex);
-            teamB.setOrderNumber(matchIndex + 1);
-
-            fantasyWeekRepository.save(fantasyWeek);
+
+        while (fantasyWeeks.size() < 38) {
```

Comment:
This shouldn't stop at 38, because if there's ten teams in the league, it will only create 8 real life weeks because there will be 5 fantasyWeeks per real life week

Reply:
you could try fantasyWeeks.size() >= 38 * matches.size() / 2

Reply:
But this could cause errors if there aren't 38 calendar weeks in a season